### PR TITLE
Simplified, unattended setup

### PR DESCRIPTION
**Pull request by flying-sausages**
This PR combines #438 and #423 with merge conflicts resolved so that we don't have to wait to merge them later

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -14,215 +14,228 @@
...
+            echo "/etc/swizzin/.dev.lock created"
+        fi
+        echo "Best of luck and please follow the contribution guidelines cheerio"
+    else
+        echo "Cloning swizzin repo to localhost"
+        git clone https://github.com/swizzin/swizzin.git /etc/swizzin >> ${log} 2>&1
+        echo -e "\tSwizzin cloned!"
+    fi
+    ln -s /etc/swizzin/scripts/ /usr/local/bin/swizzin
+    chmod -R 700 /etc/swizzin/scripts
```
Comment (liaralabs) [inline setup.sh:49]: Didn't you want to get rid of this line because it causes git to be chatty when doing `box update`s?

File: setup.sh [hunk 2]
```diff
@@ -14,215 +14,228 @@
...
+        echo_progress_done
     fi
 
-    echo "Performing a system upgrade"
-    apt-get -q -y update >> ${log} 2>&1
-    apt-get -q -y upgrade >> ${log} 2>&1
+    apt_upgrade
+
+    if ! bash /etc/swizzin/scripts/update/10-dependencies.sh; then
+        echo_error "Dependencies failed to install"
```
Comment (liaralabs) [inline setup.sh:191]: Should we consider exiting here if depends fail? In my experience the install is going to botch pretty hard if these don't get installed.

File: scripts/box [hunk 3]
```diff
@@ -304,19 +292,20 @@ function _adduser() {
...
     USER=${user^^}
-    cat > /etc/sudoers.d/${user} << SUD
+
+    #shellcheck source=sources/functions/utils
+    . /etc/swizzin/sources/functions/utils
+    cat > /etc/sudoers.d/"${user}" << SUD
 Cmnd_Alias   ${USER}CMDS = /bin/systemctl stop flood@${user}, /bin/systemctl restart flood@${user}, /bin/systemctl start flood@${user}, /bin/systemctl start rtorrent@${user}, /bin/systemctl stop rtorrent@${user}, /bin/systemctl restart rtorrent@${user}, /bin/systemctl start deluged@${user}, /bin/systemctl stop deluged@${user}, /bin/systemctl restart deluged@${user}, /bin/systemctl start deluge-web@${user}, /bin/systemctl stop deluge-web@${user}, /bin/systemctl restart deluge-web@${user}, /bin/systemctl start irssi@${user}, /bin/systemctl stop irssi@${user}, /bin/systemctl restart irssi@${user}, /bin/systemctl start qbittorrent@${user}, /bin/systemctl stop qbittorrent@${user}, /bin/systemctl restart qbittorrent@${user}, /usr/local/bin/swizzin/box chpasswd ${user}
 
 ${user}     ALL = (ALL) NOPASSWD: ${USER}CMDS
 SUD
```
Comment (liaralabs) [inline scripts/box:302]: We don't want this to run if the user is master as they get full sudo permissions

**Review by liaralabs — CHANGES_REQUESTED** [review]
So it looks like we still need unattended variables for:
- qBittorrent
- Libtorrent
- Deluge
- rTorrent

Unless these are not high priority since the installer will ask questions off the bat; however, for it to be truly unattended we should have a mechanism in place for the user to be able to skip these prompts. I don't know if we want to wait until the libtorrent overhaul takes place, but I'd like to get this PR merged sooner rather than later.

**Comment by liaralabs** [comment]
I also think we could push this to the next level by having the ability to define a post-install script

File: setup.sh (hunk 1 above)
Comment (flying-sausages) [reply setup.sh:49]: Not if we keep our permissions in check ha

File: scripts/box (hunk 3 above)
Comment (flying-sausages) [reply scripts/box:302]: Aren't these the sudo permissions to run their own tasks?

**Pull request merged**